sound/soc/sof/ipc4-topology.h

```diff
@@ -206,8 +207,8 @@ struct sof_ipc4_control_data {
 	int index;
 
 	union {
-		struct sof_ipc4_ctrl_value_chan chanv[0];
-		struct sof_abi_hdr data[0];
+		DECLARE_FLEX_ARRAY(struct sof_ipc4_ctrl_value_chan, chanv);
+		DECLARE_FLEX_ARRAY(struct sof_abi_hdr, data);
```

Comment:
Before:
```
sound/soc/sof/ipc4-control.c: note: in included file:
sound/soc/sof/ipc4-topology.h:210:40: warning: array of flexible structures

sound/soc/sof/ipc4-topology.c: note: in included file:
sound/soc/sof/ipc4-topology.h:210:40: warning: array of flexible structures
```

After:
```
sound/soc/sof/ipc4-control.c: note: in included file:
sound/soc/sof/ipc4-topology.h:211:17: warning: array of flexible structures

sound/soc/sof/ipc4-topology.c: note: in included file:
sound/soc/sof/ipc4-topology.h:211:17: warning: array of flexible structures
```
The problem is that `struct sof_abi_hdr` itself has flexible array (`__u32 data[];`), a flexible array of a flexible struct is hard to digest..

Reply:
Ok... just brainlessly applied the macros in place of [0]. My first reaction would be to change struct sof_abi_hdr data[0]; in struct sof_ipc4_control_data to simply struct sof_abi_hdr data; ... but I do not know the code that well. In any case flex array of flex arrays must be a bug, so one of the flex arrays must go.

Reply:
Well, it seems I can change "struct sof_abi_hdr data[0];" to "struct sof_abi_hdr data;" in both places without any changes anywhere... so it looks like those data members are not touched anywhere. But I am completely lost here, I have no idea what these structs are for and why they have been made flexible in the first place.... but my IPC4 probes changes has nothing to do with those and should get merged soon, so I drop everything else but the probes related changes.

Reply:
Some reason Intel related things have this love for flexible array of flexible arrays nested.

Imho it is just plain wrong to use a struct (which is a flexible struct) as a flexible member...

```c
struct sof_ipc4_control_data {
	struct sof_ipc4_msg msg;
	int index;

	union {
		struct sof_ipc4_ctrl_value_chan chanv[0];
		struct sof_abi_hdr data[0];
	};
} blah;
```
Where `blah.data[2]` is pointing?

My uneducated fixes for similar issues for NHLT: 62c03bcc4ad3b529f1d954ad705557f53c566b78, 2b79c659507d8dd00d0fe0cbec95290fd23dfe4b and one for SOF: 490217ea34e4b33566c7d360959f24b32c6c6e95.
I'm not sure if this is correct, but I don't see other way to fix it.

I can not just drop the `[]` from the end as we might have multiple flexible structs placed after each other, but having it named as a struct is not right either.